packages/client/src/user/userReducer.ts

```diff
@@ -259,6 +259,19 @@ export interface IUserFormState {
   userAuditForm: IUserAuditForm
 }
 
+function withScopes<T extends Record<string, unknown>>(
+  values: T,
+  userRoles: UserRole[]
+): T & { scopes?: string[] } {
+  if (!('role' in values)) {
+    return values
+  }
+  return {
+    ...values,
+    scopes: userRoles.find((role) => role.id === values.role)?.scopes
+  }
+}
```

Comment:
So the problem with scopes is often that even though the current user's scopes are easy to find from JWT, other people's scopes isn't so simple, and sometimes we have views which use not-your-scope like `SCOPES.PROFILE_ELECTRONIC_SIGNATURE`

Reply:
in this case, we are resolving the scopes of the user that is getting created. The signature field's appearance depends on that

Reply:
Yeah makes sense. Bit of an edge case to need to figure out someone else's scopes, or did you find it common during the scope work? Just curious for future.

Reply:
Hm had encountered it a couple of times but both of them were related to the user creation form